### PR TITLE
Added power-ups to Skeletons ad Wariors

### PR DESCRIPTION
**Pull request by AlecsFerra**

### CODE REVIEW

File: Enemy.java [hunk 1]
```diff
@@ -75,8 +75,8 @@ public String name()
      * 
      * @return heatlh of this enemy
      */
-    public int health()
-    {
+    public int jealth() {
```
Comment (maaslalani) [inline Enemy.java]: `jealth`?

**Comment by maaslalani** [comment]
@AlecsFerra I'm confused about `jealth` is that a typo? Otherwise code looks really good! Thank you very much :)
Do you think it would make sense to create `Enemy` subclasses like with `Skeleton` and `Warrior` so that we don't get a bunch of `if / else` statements in the codebase?

**Comment by AlecsFerra** [comment]
That was a typo, for the other things i suggest an enum

**Comment by maaslalani** [comment]
We are straying away from simply checking the `name` of the `enemy` and then applying special effects. The solution we are going with is to have an abstract `enemy` class with all subclasses inheriting and overriding their `attack` and `takeDamage` methods, we will also move a `steal` method here for when the player decides to `runAway`.

**Comment by maaslalani** [comment]
@AlecsFerra Checkout the master branch of Dungeon, and try implementing the changes again, the `enemy` superclass and abstract classes have been setup.

**Pull request closed without merging**